tools/pd-heartbeat-bench/main.go

```diff
@@ -333,11 +330,32 @@ func (rs *Regions) handleRegionHeartbeat(wg *sync.WaitGroup, stream pdpb.PD_Regi
 	log.Info("store finish one round region heartbeat", zap.Uint64("store-id", storeID), zap.Duration("cost-time", time.Since(start)))
 }
 
-func (rs *Regions) collectStoresStats(storeCount int) []*pdpb.StoreStats {
-	stores := make([]*pdpb.StoreStats, storeCount+1)
+// Stores contains store stats with lock.
+type Stores struct {
+	sync.RWMutex
+	stat []*pdpb.StoreStats
+}
+
+func newStores(stores int) *Stores {
+	return &Stores{
+		stat: make([]*pdpb.StoreStats, stores+1),
+	}
+}
+
+func (s *Stores) heartbeat(ctx context.Context, cli pdpb.PDClient, storeID uint64) {
+	s.RLock()
+	defer s.RUnlock()
+	cctx, cancel := context.WithCancel(ctx)
+	defer cancel()
+	cli.StoreHeartbeat(cctx, &pdpb.StoreHeartbeatRequest{Header: header(), Stats: s.stat[storeID]})
+}
+
+func (s *Stores) update(rs *Regions) {
+	s.Lock()
```

Comment:
Will this lock be bottleneck in large number regions benchmark?

Reply:
Their result is similar with 2000000 regions